### PR TITLE
Fix minor issues reported by IntelliJ IDEA MODPERMS-131

### PR DESCRIPTION
**Pull request by adamdickmeiss**
Should we import okapi-common and also get rid of untyped Future(s), Promise(s)?

### CODE REVIEW

**Comment by craigmcnally** [comment]
Yes I think we were waiting until we upgraded to a newer version of rmb.

File: src/test/java/org/folio/permstest/TestUtil.java [hunk 1]
```diff
@@ -72,8 +59,8 @@ public JsonObject getJson() {
       .putHeader("content-type", CONTENT_TYPE_JSON)
       .putHeader("accept", CONTENT_TYPE_JSON);
     if (headers != null) {
-      for (Map.Entry entry : headers.entries()) {
-        request.putHeader((String) entry.getKey(), (String) entry.getValue());
+      for (Map.Entry<String, String> entry : headers.entries()) {
+        request.putHeader(entry.getKey(), entry.getValue());
```
Comment (julianladisch) [inline src/test/java/org/folio/permstest/TestUtil.java:63]: Can you use HttpRequest#putHeaders(MultiMap) ?
https://github.com/vert-x3/vertx-web/blob/4.0.0/vertx-web-client/src/main/java/io/vertx/ext/web/client/impl/HttpRequestImpl.java#L160-L164

File: src/main/java/org/folio/rest/impl/PermsAPI.java [hunk 2]
```diff
@@ -193,31 +193,30 @@ void postPermsUsersTrans(PermissionUser entity, Context vertxContext,
...
+                                  ID_FIELD, permUser.getId(),
+                                  UNABLE_TO_UPDATE_DERIVED_FIELDS + cause.getMessage()))));
+                    } else {
+                      asyncResultHandler.handle(Future.succeededFuture(
+                          PostPermsUsersResponse.respond500WithTextPlain(
+                              cause.getMessage())));
+                    }
+                  })
+              )
+              .onSuccess(res1 ->
```
Comment (hjiebsco) [inline src/main/java/org/folio/rest/impl/PermsAPI.java:214]: Maybe `res` not `res1`?

File: src/test/java/org/folio/rest/impl/PermsAPITest.java [hunk 3]
```diff
@@ -95,7 +95,7 @@ public void testUpdateUserPermissionsFail(TestContext context) {
     Context vertxContext = vertx.getOrCreateContext();
     PostgresClient postgresClient = PostgresClient.getInstance(vertxContext.owner(), tenantId);
     postgresClient.startTx(s -> {
-      Future<Void> future = PermsAPI.updateUserPermissions(s, "bad",
+      Future<Void> future = api.updateUserPermissions(s, "bad",
```
Comment (hjiebsco) [inline src/test/java/org/folio/rest/impl/PermsAPITest.java:98]: `updateUserPermissions` is a static method. Consider to use `PermsAPI.updateUserPermissions` but remove the unnecessary line 92 `PermsAPI api = new PermsAPI();`

File: src/main/java/org/folio/rest/impl/PermsAPI.java (hunk 2 above)
Comment (adamdickmeiss) [reply src/main/java/org/folio/rest/impl/PermsAPI.java:214]: yep

File: src/test/java/org/folio/permstest/TestUtil.java (hunk 1 above)
Comment (adamdickmeiss) [reply src/test/java/org/folio/permstest/TestUtil.java:63]: yes,

File: src/test/java/org/folio/rest/impl/PermsAPITest.java (hunk 3 above)
Comment (adamdickmeiss) [reply src/test/java/org/folio/rest/impl/PermsAPITest.java:98]: ok

**Review by hjiebsco — APPROVED** [review]

**Pull request merged**